youtube_dl/extractor/seeso.py

```diff
@@ -0,0 +1,91 @@
+    _VALID_URL = r'https?://(?:www\.)?seeso\.com/view/episode/(?P<id>[0-9]+)'
+    _TEST = {
+        'params': {
+            'username': 'emailhere',
+            'password': 'passwordhere'
+        },
+        'url': 'https://www.seeso.com/view/episode/799241283849',
+        'info_dict': {
+            'id': '799241283849',
+            'ext': 'mp4',
+            'series': 'Bajillion Dollar Propertie$',
+            'title': 'Farsi Lessons',
+            'description': ("Amir leads Victoria into a trap, the Bros meet a super obnoxious bro, "
+                            "and rival brokers Serge and Gio intimidate Glenn."),
+            'thumbnail': 'https://chaosic.akamaized.net/NBCOTT_-_Production/360/907/160831_3092487_Farsi_Lessons.jpg',
+        }
+    }
+
+    def _real_extract(self, url):
+        username, password = self._get_login_info()
+        video_id = self._match_id(url)
+
+        if username is None or password is None:
+            return
+
+        _API_AUTH_URL = "https://www.seeso.com/api/v3/auth/login"
+        auth_json = {
+            "username": username,
+            "password": password,
+            "userOnly": True,
+            "isRememberMe": True,
+            "isWeb": True
+        }
+
+        # Send auth POST request and get token from response
+        auth_request = sanitized_Request(_API_AUTH_URL, urlencode_postdata(auth_json))
+        auth_response = json.loads(self._download_webpage(auth_request, '', note='Getting auth token...'))
+        auth_token = auth_response.get('user').get('token')
+
+        # Use the public unauthenticated API to get the video's info
+        _VIDEO_INFO_URL = 'https://feed.theplatform.com/f/NZILfC/nbcott-prod-all-media?byAvailabilityState=' \
+                          'available&byId=%s&form=cjson'
+        request = sanitized_Request(_VIDEO_INFO_URL % video_id)
+        json_data = self._download_json(request, video_id)
+        entry = json_data.get('entries')[0]
+
+        # Template fields
+        series = entry["nbc-chaos$show"]                    # Show name
+        title = entry["nbc-chaos$shortTitle"]               # Episode Name
+        season_number = entry["nbc-chaos$seasonNumber"]
+        episode_number = entry["nbc-chaos$episodeNumber"]
+        thumbnail = entry["defaultThumbnailUrl"]
+        description = entry["nbc-chaos$shortDescription"]
+
+        # Got the show's public URL. Now we need to parse out the videoID
+        public_url_id = os.path.split(urlparse.urlparse(entry["publicUrl"]).path)[-1]
+
+        # Get the master m3u8 which lists formats
+        m3u8_url = 'https://link.theplatform.com/s/NZILfC/media/{0}?feed=All%20Media%20Feed&auth={1}' \
+                   '&vpaid=script,flash&formats=m3u,mpeg4'.format(public_url_id, auth_token)
+        formats = []
+        for entry in self._extract_m3u8_formats(m3u8_url, video_id, m3u8_id='m3u8', ext='mp4'):
+            formats.append(entry)
+
+        self._sort_formats(formats)
```

Comment:
I've been banging my head against the wall trying to get the existing ThePlatform methods to work to no avail. I've pushed the changes I've made, but it's failing right now because one of the methods thinks there's no auth token. 

I don't really understand how my original way of doing it was worse than these methods. It was much shorter and cleaner.

<pre>
[Seeso] : Getting auth token...
extract_video_info - auth_token:  cKgTqEE0BuwgJjg2qcWeAAAWEIB2YEDa
(u'[ThePlatformFeedIE] real_url: ', u'https://feed.theplatform.com/f/NZILfC/nbcott-prod-all-media?form=json&byId=799241283849')
[Seeso] <function <lambda> at 0x11136fd70>: Downloading JSON metadata
[Seeso] <function <lambda> at 0x11136fd70>: Downloading SMIL data for Video
ERROR: This content requires a valid, unexpired auth token.
Traceback (most recent call last):
  File "/Volumes/750DATA/Projects/youtube-dl/youtube_dl/YoutubeDL.py", line 694, in extract_info
    ie_result = ie.extract(url)
  File "/Volumes/750DATA/Projects/youtube-dl/youtube_dl/extractor/common.py", line 357, in extract
    return self._real_extract(url)
  File "/Volumes/750DATA/Projects/youtube-dl/youtube_dl/extractor/seeso.py", line 90, in _real_extract
    return self._extract_video_info(video_id, auth_token)
  File "/Volumes/750DATA/Projects/youtube-dl/youtube_dl/extractor/seeso.py", line 41, in _extract_video_info
    'auth': auth_token,
  File "/Volumes/750DATA/Projects/youtube-dl/youtube_dl/extractor/theplatform.py", line 335, in _extract_feed_info
    smil_url, query), video_id, 'Downloading SMIL data for %s' % asset_type)
  File "/Volumes/750DATA/Projects/youtube-dl/youtube_dl/extractor/theplatform.py", line 40, in _extract_theplatform_smil
    raise ExtractorError(error_element.attrib['abstract'], expected=True)
ExtractorError: This content requires a valid, unexpired auth token.

E
======================================================================
ERROR: test_Seeso (__main__.TestDownload)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_download.py", line 142, in test_template
    force_generic_extractor=params.get('force_generic_extractor', False))
  File "/Volumes/750DATA/Projects/youtube-dl/youtube_dl/YoutubeDL.py", line 709, in extract_info
    self.report_error(compat_str(e), e.format_traceback())
  File "/Volumes/750DATA/Projects/youtube-dl/youtube_dl/YoutubeDL.py", line 568, in report_error
    self.trouble(error_message, tb)
  File "/Volumes/750DATA/Projects/youtube-dl/youtube_dl/YoutubeDL.py", line 538, in trouble
</pre>